
% Version 1.0 January 2009
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template

\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color} 

% Use doublespacing - comment out for single spacing
\usepackage{setspace} 
\doublespacing


% Text layout
\topmargin 0.0cm
\oddsidemargin 0.5cm
\evensidemargin 0.5cm
\textwidth 16cm 
\textheight 21cm

% Bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}

% Use the PLoS provided bibtex style
\bibliographystyle{plos2009}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother


% Leave date blank
\date{}

\pagestyle{myheadings}
%% ** EDIT HERE **


%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW
\usepackage{pifont}
\usepackage{hyperref}
\hypersetup{
    colorlinks=false,
    pdfborder={0 0 0},
}

\newcommand{\diff}[1]{\textcolor{blue}{#1}}  % highlights using blue, to indicate things that have changed
\newcommand{\note}[1]{\textcolor{red}{[#1]}}  % notes are included in brackets and in red

\newcommand{\mul}{\ensuremath{\mathrm{\mu l}}}
\newcommand{\C}{\,$^{\circ}\mathrm{C}$}
\usepackage{tablefootnote}
\usepackage[symbol*]{footmisc}

% Thanks, stackexchange! :)
\DefineFNsymbolsTM{myfnsymbols}{% def. from footmisc.sty "bringhurst" symbols
  \textdagger    \dagger
  \textasteriskcentered *
  \textdaggerdbl \ddagger
  \textsection   \mathsection
  \textbardbl    \|%
  \textparagraph \mathparagraph
}%
\setfnsymbol{myfnsymbols}


%% END MACROS SECTION

\begin{document}

% Title must be 150 characters or less
\begin{flushleft}
{\Large
\textbf{PAT-seq --- a whole-transcriptome poly(A) tail length determination assay for the Illumina platform}
}
% Insert Author names, affiliations and corresponding author email.
\\
\vspace{5mm}
Botond Sipos$^{1, \ast}$, 
Adrian M. St\"utz$^{2}$, 
Greg Slodkowicz$^{1}$,
Tim Massingham$^{1}$\footnote{Current address: Oxford Nanopore Technologies, Edmund Cartwright House, 4 Robert Robinson Avenue, Oxford Science Park, Oxford, OX4 4GA, United Kingdom.},
Jan Korbel$^{2}$, 
Nick Goldman$^{1}$
\\
\vspace{5mm}
$^1$European Molecular Biology Laboratory, European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge, United Kingdom
\\
$^2$European Molecular Biology Laboratory, Genome Biology Unit, Heidelberg, Germany 
\\
$^\ast$E-mail: sbotond@ebi.ac.uk
\\
\vspace{5mm}
\textbf{Keywords:} transcriptomics; RNA-seq; poly(A) tail; PAT; high-throughput assay; PAT-seq

\end{flushleft}

\section*{Abstract}

The poly(A) tail, co-transcriptionally added to most eukaryotic mRNAs, plays an important role in post-transcriptional regulation through modulating mRNA stability and translational efficiency. The length of the poly(A) tail is dynamic, decreasing or increasing in response to various stimuli through the action of enzymatic complexes, and changes in tail length are exploited in regulatory pathways implicated in various biological processes.

To date, assessment of poly(A) tail length has mostly relied on protocols targeting only a few transcripts. We present PAT-seq (`poly(A) tail length determination by sequencing'), a whole-transcriptome approach to measure tail lengths  --- including a computational pipeline implementing all necessary analyses. PAT-seq uses direct Illumina sequencing of cDNA fragments obtained through G-tailing of poly(A)-selected mRNA followed by fragmentation and reverse transcription. 

Analysis of reads corresponding to spike-in poly(A) tracts of known length indicated that mean tail lengths can be confidently measured, given sufficient coverage. We further explored the utility of our approach by comparing tail lengths estimated from wild type and \textit{$\Delta$ccr4-1/pan2} mutant yeasts. The yeast whole-transcriptome tail length distributions showed high consistency between biological replicates, and the expected upward shift in tail lengths in the mutant samples was detected. This suggests that PAT-seq is suitable for the assessment of global polyadenylation status in yeast.

The correlation of per-transcript mean tail lengths between biological and technical replicates was low (higher between mutant samples). Both, however, reached high values after filtering for transcripts with greater coverage. We also compare our results with those of other methods. We identify a number of improvements that could be used in future PAT-seq experiments and, based on our results, believe that direct sequencing of poly(A) tails can become the method of choice for studying polyadenylation using the Illumina platform.

\section*{Introduction}

Polyadenylation --- the cleavage of the nascent mRNA and the addition of a tract consisting of adenine bases in a non-template-dependent manner --- is a post-transcriptional modification characteristic of the majority of eukaryotic transcripts \cite{zhao99,elkon13,tian13}. It is performed by a well-characterised protein complex (reviewed in \cite{zhao99}) initiated by a main poly(A) site signal and modulated by various upstream and downstream regulatory elements (reviewed in \cite{tian11,weill12}). Both the presence of the poly(A) tail and variation of its length in response to various stimuli are important in numerous biological processes.

There can be several main signal elements within a single transcript defining distinct poly(A) sites (PASs) with different affinity \cite{tian13,elkon13}. Since the PASs define the actual end of the transcript, differential PAS usage contributes to the processes increasing the complexity of eukaryotic transcriptomes. The differential usage of alternative PASs under different conditions is modulated by various protein factors and chromatin structure \cite{tian13}, and as such it is a part of the repertoire of post-transcriptional regulatory mechanisms of gene expression along with splicing \cite{licatalosi10}. The length of the poly(A) tail (PAT) itself also plays an important role in post-transcriptional regulation of gene expression and various cellular processes such as mRNA export \cite{fuke07} and stability \cite{lackner07,eckmann11,weill12}, translational efficiency \cite{weill12}, cell cycle regulation \cite{mendez01} and microRNA action \cite{moretti12}.

The poly(A) tails are bound by poly(A) binding proteins (PABs) during their synthesis in the nucleus. After export to the cytoplasm the PABs associate with the eIF4-G translation initiation factor binding the $5'$ 7-methylguanylate cap (m$^7$G). This creates a circular structure which protects the mRNA from the cytoplasmic exonucleases, giving rise to the well-established effect of the tail length on mRNA stability \cite{subtelny14,houseley06}. The m$^7$G and the PAT also synergistically increase translational efficiency \cite{gallie91}, further emphasizing the role of the circular structure. While the evidence for the enhancing effect of the PAT on translational efficiency is ample \cite{preiss98,lackner07}, the recent high-throughput study by \cite{subtelny14} suggests that the role the tail length plays in the regulation of translational efficiency is limited to certain cellular contexts, such as early embryogenesis. For example, \cite{subtelny14} found no correlation between tail length and translational efficiency in \textit{Saccharomyces cerevisiae}.

Through its role in the cellular processes above, changes in the length of the PAT are involved in a range of important biological processes such as inflammation \cite{weill12}, embryogenesis \cite{weill12,subtelny14}, axonal transport \cite{weill12}, circadian rhythm \cite{gotic12} and DNA damage response \cite{traven05}. The initial length ($\sim$250 bases in humans, $\sim$70--80 in yeast according to \cite{eckmann11}) to which the poly(A) polymerase extends the tail is regulated by the affinity and homopolymer structure of the PABs. After synthesis, the tail length is modified by various adenylases and deadenylases, with the enzymes and mechanisms involved showing considerable variability between different organisms \cite{brown98,eckmann11}. In the case of \textit{S.~cerevisiae}, the tail is tailored to a message-specific length by the nuclear PAN deadenylase complex, governed by regulatory elements located in the $3'$-UTR \cite{brown98,beilharz07}. Throughout the lifetime of the mRNA in the cytoplasm the tail is gradually shortened by the Ccr4-Not complex. Due to this and similar processes, the tail length in total mRNA usually differs from the default length (median tail length 75--100 in human; mean 33 in yeast according to \cite{subtelny14}).

The genome-wide study of alternative polyadenylation has been stimulated by the development of assays based on high-throughput sequencing (reviewed in \cite{elkon13}). Regular RNA-seq is inefficient for the study of PASs and PATs, as only a small fraction of the reads overlap with these regions for reasons that are not fully understood \cite{fu11,wilkening13}. Consequently, assays have been developed that compensate for this through enrichment techniques. An additional complication is that on the Illumina sequencing platform reads overlapping PATs have low quality values \cite{wilkening13}, suggestive of unreliable base calls. Likely causes of this issue are mispriming of the sequencing oligo and increased phasing noise due to polymerase slippage \cite{wilkening13}. Both issues are related to the low sequence complexity of the tail; hence some protocols, such as SAPAS \cite{fu11}, introduce random mutations in the sequencing library in order to mitigate this.

\note{some redundant text removed here}
% Since the above issues have a large impact on the measurement of PAT lengths through direct sequencing, 
The development of high-throughput PAT length assays has been slower than the development of protocols targeting PASs. Until the recent publication of the PAL-seq method by \cite{subtelny14}, high-throughput poly(A) assays were based on fractionation of mRNAs according to the tail length using affinity chromatography followed by microarray analysis \cite{beilharz07,meijer07}. Hence, most of the literature regarding PAT length relates to low-throughput experiments using various techniques targeting specific transcripts \cite{salles95} and based on reverse transcription and PCR using transcript-specific and tail-specific primers. The tail-specific primers can be simple oligo dTs (as in the RACE-PAT assay \cite{salles95}) or primers targeting tags at the end of the tail engineered through ligation (LM-PAT \cite{salles95}), extension by the Klenow fragment (E-PAT \cite{janicke12}), or G-tailing \cite{kusov01}. These methods are well-established; however, they are impractical for large-scale studies.

The high-throughput PAL-seq assay \cite{subtelny14} is a customized experiment on the Illumina sequencing platform involving a ``T-filling'' step similar to \cite{wilkening13} using a mixture of dTTP and biotinylated dUTP and measurement of fluorescence of labeled streptavidin bound to the clusters. After normalization based on signal from spike-ins with poly(A) tracts of known length, the measured fluorescence is converted into estimated tail lengths. The PAL-seq assay has good reproducibility (with a correlation of 0.93 between biological replicates), but considering the complexity of the protocol there is still advantage to be gained from alternative high-throughput PAT length assays, preferably based on direct sequencing of the tail.

In this study we explore PAT-seq, an alternative high-throughput Illumina-based PAT assay based on the (low-throughput) G-tailing approach of \cite{kusov01}. In brief, our approach is based on G-tailing poly(A)-selected mRNA followed by fragmentation and reverse transcription using a mixture of primers targeting the poly(A)/G-tag junction and the transcript-end/poly(A) transition. The resulting cDNA fragments are sequenced using a standard Illumina RNA-seq protocol (see \textbf{Figure~1}).

We validate PAT-seq by experiments performed on wild type (WT) \textit{S.~cerevisiae} and \textit{$\Delta$ccr4-1/pan2} mutants. In addition to yeast being the most popular model organism for studying polyadenylation and developing novel PAT assays, it is known that the \textit{$\Delta$ccr4-1/pan2} mutant has more homogeneous and longer tails which better reflect the default size after synthesis in the nucleus \cite{beilharz07,traven05}. We use the prior expectation this establishes regarding both the direction and magnitude of shift in tail length when compared to the WT to assess the performance of PAT-seq. We also compare PAT-seq to the state-of-the art Illumina-based (PAL-seq) and microarray-based (PASTA) assays.

\section*{Materials and methods}

\subsection*{RNA isolation}

WT yeast (BY4741, which is derived from S288c) and \textit{$\Delta$ccr4-1/pan2} \cite{beilharz07} were single colony-streaked on YPAD-agar plates, and two individual colonies each were inoculated in 2ml YPAD medium and grown o.n. at 30 at 180rpm. OD was measured and 7$\times$10$^7$ cells were pelleted, resuspended in 2ml Y1 medium (2{\mul} beta-ME, 150{\mul} zymolase) and incubated for 30min at 30{\C} in a waterbath. After centrifugation, the pellet was lysed in buffer RLT and processed according to the yeast protocol of the Qiagen RNeasy kit. Yield and quality of total RNA was measured with Nanodrop and confirmed by Agilent RNA Nano chip (RIN 10). A DNase digestion was performed on 20$\mathrm{\mu g}$ totalRNA using Turbo DNase (Life Technologies), followed by an RNeasy column cleanup step and elution into 30{\mul} nuclease free water (Ambion).

\subsection*{Poly(A) selection}

Two rounds of poly(A) selection were performed using the TruSeq RNA Sample Preparation kit (Illumina) with several modifications. In detail: 50{\mul} of DNase-digested totalRNA was incubated with 50{\mul} RNA Purification beads (oligo-dT) for 5min at 65{\C}, cooled to 4{\C}, and then incubated for 5min at RT in a DNA Engine tetrade 2 (Biorad) thermocycler. Afterwards, beads were separated with a magnet and washed with 200{\mul} Bead Wash buffer, and then eluted in 50{\mul} elution buffer for 2min at 80{\C}. On RT, 50{\mul} of binding buffer were added, placed on a magnet and washed with 200{\mul} Bead Wash buffer, and again eluted in 50{\mul} elution buffer for 2min at 80{\C}. The supernatant containing the mRNA was transferred into a 1.5ml Low Binding tube (Eppendorf) and quantified with the Qubit RNA assay.

\subsection*{Spike-in control}

20{\mul} of spike-in \textit{in vitro} transcripts with a known A$_{42}$ stretch \cite{pelechano13} were added. 
\note{If it's A$_{42}$, how come everywhere else we now state that the expected A length is 43?}
A cleanup step was performed using the RNeasy MinElute Cleanup Kit (Qiagen) according to protocol, eluting with 14{\mul} water.

\subsection*{G-tailing and fragmentation}

The USB Poly(A) Tail-Length Assay kit (Affymetrix) was used with several modifications to add a short G stretch to the $3'$ end of the mRNA. In detail, 14{\mul} mRNA (with spike-in) were incubated with 4{\mul} of 5x Tail buffer mix and 2{\mul} 10x Tail Enzyme mix at 37{\C} for 1h and stopped by addition of 2{\mul} 10x Tail stop solution. After an RNeasy Minelute cleanup step and elution into 18{\mul} of water, chemical fragmentation using the NEBnext Magnesium RNA Fragmentation kit (NEB) was performed in 20{\mul} for 3.5min at 94{\C}. Afterwards, the reaction was immediately put on ice, and 2{\mul} of RNA Fragmentation Stop Solution was added. Another RNeasy Minelute cleanup step was performed, eluting into 15{\mul} of water.

\subsection*{cDNA synthesis}

1{\mul} TTTTVN (16.7$\mathrm{\mu M}$) and 1{\mul} CCCCCCTT (50$\mathrm{\mu M}$) custom primers (both Sigma) were added to the 15{\mul} fragmented RNA, primed by incubation at 65{\C} for 5min and put on ice. 1$^\mathrm{st}$ strand cDNA synthesis was performed using the 1$^\mathrm{st}$ strand mastermix (Illumina TruSeq RNA kit) and SuperscriptII enzyme (Life Technologies) by incubating at 25{\C} for 10min, 42{\C} for 50min and 70{\C} for 15min, and cooled to 4{\C} in a MJ-Mini thermocycler (Biorad). 2$^\mathrm{nd}$ strand synthesis was performed by addition of 25{\mul} 2$^\mathrm{nd}$ strand master mix and incubation for 1h at 16{\C}.

\subsection*{NGS library preparation}

An AMPure XP (Beckman Coulter) cleanup step was performed by adding 90{\mul} AMPure XP beads, vortexing and incubating for 15min at RT. After separation on a magnet, the supernatant was removed and the beads were washed twice with 200{\mul} 80\% ethanol. After air drying for 10min at RT, cDNA was eluted in 50{\mul} Resuspension buffer and stored at -20{\C}.
End repair was performed by adding 10{\mul} Resuspension buffer and 40{\mul} Endrepair mix and incubating at 30{\C} for 30min at 750rpm in a thermomixer (Eppendorf).
After another AMPure XP cleanup step using 160{\mul} beads and eluting into 17.5{\mul} Resuspension buffer, the A-overhang addition step was performed. 12.5{\mul} A-tailing mix were added and incubated at 37{\C} for 30min at 750rpm, followed by 5min at 70{\C} and put on ice. Adapter ligation used 2.5{\mul} of a 1:40 dilution of Adapter oligo mix (Agilent Sureselect v4 kit), 2.5{\mul} Resuspension solution and 2.5{\mul} DNA ligase mix, incubated at 30{\C} for 10min in a water bath and stopped by addition of 5{\mul} Stop ligase mix. The adapter-ligated cDNA library was cleaned up by two rounds of AMPure XP using first 42{\mul} of AMPure XP beads and eluting in 50{\mul} Resuspension buffer and then using 50{\mul} AMPure XP beads and eluting in 40{\mul} Resuspension buffer.

Next, a size selection step was performed to create four technical replicates between 300--325bp, 325--350bp, 400--425bp and 425--450bp. For this, the purified adapter ligated cDNA was loaded on a 1.5\% Agarose gel and separated for 2h at 120V. Using SybrSafe (Invitrogen) as dye and a Dark Reader transilluminator (Clare Chemical Research), narrow bands were cut using 100bp marker as guidance and gel extracted using the Nucleospin Gel and PCR Cleanup kit (Macherey-Nagel) and eluted in 22{\mul} NE buffer.
Two sequential rounds of PCR were performed to introduce barcodes and amplify the cDNA library. First, the Sureselect primer and Sureselect ILM Indexing Pre-Capture PCR reverse primer were used with the Phusion HF 2x mix in 50{\mul} in a MJ Mini thermocycler as follows: 2min 98{\C}, followed by 5 cycles of 30sec 98{\C}, 30sec 65{\C}, 1min at 72{\C}, and a final 72{\C} step for 10min. Next, an AMPure XP cleanup step was performed using 90{\mul} AMPure XP beads and eluting into 23{\mul} water. Afterwards, a 2$^\mathrm{nd}$ PCR was performed using the Sureselect ILM Indexing Post Capture Forward PCR primer and a different PCR Primer Index primer/sample together with the Phusion HF 2x mix in 50{\mul} in a MJ Mini thermocycler as follows: 2min 98{\C}, followed by 13--15 cycles of 30sec 98{\C}, 30sec 57{\C}, 1min at 72{\C}, and a final 72{\C} step for 10min. After a final AMPure XP cleanup step using 90{\mul} beads, the final library was eluted in 20{\mul} water, quantified with the Qubit HS dsDNA kit and pooled equimolarly.

\subsection*{Illumina sequencing}

All 16 pooled barcoded libraries were sequenced on one HiSeq2000 lane in 2 $\times$ 101bp mode. Raw data has been deposited to Array Express under accession number \href{http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2456/}{E-MTAB-2456}.

\subsection*{Data analysis}

The data analysis pipeline with documentation, logs and selected raw results (see also \textbf{Text S1}) is available at \href{http://github.com/sbotond/paper-pat-seq}{http://github.com/sbotond/paper-pat-seq}. Individual primary data analysis steps implemented in specialised tools written in Python (under directory \texttt{patsy/})  and downstream data analysis steps implemented in R scripts (under \texttt{scripts/}) can be invoked through documented \texttt{make} targets.

The first primary data analysis step is the classification of read pairs into ``G-tail'' pairs, based on the presence of a sequence motif characteristic to the poly(A) tail/oligo(G)-tag junction in the first 14 aligned bases of one of the reads, or ``NVTR'' read pairs otherwise. For G-tail reads we require a minimum G-tag length of 3 and we allow for a maximum of 6 Ns in the first 14 bases. The G-tail pairs represent those that will, ideally, contain the complete PAT in one read and mappable sequence in the paired read that will unambiguously identify the transcript. We map the G-tail read pairs to the SGD \textit{S.~cerevisiae} reference transcriptome \cite{engel13} using Bowtie2 \cite{langmead12} with parameters \texttt{--ignore-quals --sensitive -I 0 -X 500 –no-contain}, chosen to perform a stringent alignment of the read anchoring the fragment to the respective transcript. 

All non-G-tail read pairs are classified as NVTR reads, representing those targeting the transcript-end/poly(A) junction as well as ``noise'' reads most likely originating from fragments that are products of off-target priming. The NVTR reads 
\note{I have tried to simplify this para.\ up to here.  Please check that I have not damaged the meaning.} 
are used to assess the insert size distributions of the technical replicates and other quality control measures. They are mapped to the SGD transcriptome using Bowtie2 with parameters \texttt{--ignore-quals --sensitive-local -I 0 -X 500 --no-mixed --no-discordant –no-contain}, aiming for a less stringent alignment of both reads corresponding to the fragment. The classification and mapping analysis steps are implemented in the \texttt{patsy-align} tool. 

The resulting G-tail and NVTR alignments are parsed into G-tail and NVTR fragments, respectively, using the \texttt{patsy-parse} tool. For each transcript, we tabulate the distribution of the number of A bases until the first non-A base after the G-tag in the G-tail reads overlapping the PAT (the ``tail run'' distribution). Insert size distributions are assessed based on the mapped NVTR fragments.

Using the \texttt{patsy-spike} tool we extract the reads mapped to the spike-in standards, identify the reads overlapping the poly(A) tracts with known lengths and plot the distributions of number of A bases until the first 1--5 non-A bases.  
Comparison of the results obtained for this range of thresholds, necessary because of the low quality scores of many reads, led us to choose a criterion of 1 non-A base for defining the PAT length.

The last step of the primary data analysis is to plot the per-transcript and global tail run distributions from wild type (WT1 and WT2) and mutant (MUT1 and MUT2) samples and to test the significance of differences using Mann-Whitney U-test (with a minimum sample size of 30).

Downstream analysis steps (invoked through other analysis pipeline \texttt{make} targets) perform filtering by G-tail fragment coverage, plotting of the correlation between biological replicates and technical replicates of the size selection step (above) as a function of different coverage thresholds, plotting and clustering of the per-transcript tail run distributions and correlation of thresholded per-transcript mean tail run lengths with results generated by the PAL-seq \cite{subtelny14} and PASTA \cite{beilharz07} assays (see the pipeline documentation).

\section*{Results and discussion}

\subsection*{Characteristics of aligned fragments}

The number of NVTR fragments was consistently 2--7 times higher than the number of G-tail fragments in all technical replicates. Also, the majority of the reads overlapping the poly(A) tails were sequenced as second reads, indicating that there were issues with detecting clusters having the repetitive tails as first reads (see alignment reports listed in \textbf{Text~S1}).

The NVTR fragments also had a higher mapping rate (49--66\%) than the G-tail fragments (26--39\%).  NVTR fragments mapped to various locations within the transcripts (see the report files listed in \textbf{Text~S1}), and for that reason we did not further investigate whether these fragments reliably define individual polyadenylation sites.

\subsection*{Quantification of tail run ``slippage'' using spike-ins}

The poor quality of reads overlapping the poly(A) tails and the ``drift'' of the sequenced tail length is a known issue \cite{wilkening13} which can be clearly observed in the reads mapping to the poly(A) tracts in our spike-ins (see the \textit{error\_thr.pdf} file 
\note{$\leftarrow$ It's not clear what this refers to.  Either make this filename visible in the Text S1 document, or else add a phrase here that indicates exactly which link in the document is meant}
linked from Text S1). 
However, the mean tail run length average 
\note{``mean \ldots average''? Is it really a mean of averages?  Even if it is, maybe it is OK to omit one of those words?}
from these 53558 reads is close to the true length of the poly(A) tracts (44.56 $\pm$ 16.11 cf.\ 43; see \textbf{Figure~2}). This suggests that given enough coverage of G-tail reads there is sufficient signal for characterising the tail length, as with increasing sample size the mean of the tail run distribution (including slippage noise) gets closer to the true tail length. The variance of the slippage noise distribution, however, also implies that filtering out transcripts with low G-tail fragment coverage is an essential step in quality control of PAT-seq data.

\subsection*{Global shift in polyadenylation status in \textit{$\Delta$ccr4-1/pan2} mutants}

The distribution of tail run lengths tabulated from the whole WT and mutant transcriptomes (\textbf{Figure~3A}) showed high consistency between biological replicates (compare \textbf{Figure~3} with \textbf{Figure~S1}). Both distributions are unimodal with a fat tail to the right. The right tail of the distribution is truncated in the case of the mutant samples as the largest measurable tail run is limited by the read length; the much lesser degree of truncation for the WT samples suggests that the read length of 101bp is capturing the full length of virtually all their poly(A) tails.

We consistently observe an upward shift in the tail length in the \textit{$\Delta$ccr4-1/pan2} mutants: in WT samples the global mean of tail run lengths is consistently around 46, which is approximately in line with values reported by other studies (median tail length of 33 reported by \cite{subtelny14}), while in the mutants the mean length is around 71. Moreover, the global shift in tail run length (+25) is consistent with the mean of per-transcript tail run shifts (compare \textbf{Figure~3B} with \textbf{Figure~3A}). These observations are in accord with expectations, given that the mutant is supposed to have longer tails ($>$55 according to \cite{beilharz07}) due to the lack of nuclear and cytoplasmic deadenylase activity. This suggests that PAT-seq is suitable for the assessment of global polyadenylation status in yeast.

\subsection*{Coverage-dependent consistency of per-transcript tail run distributions}

Despite the consistent global tail run distributions, the correlation of per-transcript mean tail run lengths between biological replicates is not high, both in the case of wild type and mutant samples (WT1 vs.\ WT2: $r$=0.25; MUT1 vs.\ MUT2: $r$=0.46).
However, the correlation increases after applying coverage filtering with increasing thresholds. After filtering for minimum G-tail coverage of 400 fragments, we are left with 1211 transcripts (out of 3951 detected overall) in wild type and 673 (out of 4646 detected) in the mutant samples, with per-transcript mean tail run correlations of 0.54 and 0.95, respectively.
At a coverage threshold of 1000 the correlation reaches high values both between wild type and mutant samples (WT1 vs.\ WT2: r=0.73; MUT1 vs.\ MUT2: r=0.98; \textbf{Figure 4A}), but we lose a greater proportion of the detected transcripts (465 transcripts retained in the WT samples; 315 transcripts retained in the mutant). 
This behavior is consistent with the observations drawn above from the analysis of spike-in reads, and also explains the consistency of the global tail run distributions which are necessarily dominated by more abundantly expressed transcripts.

\note{Deleted further repetition of the coverage threshold 1000 results}
%At a coverage threshold of 1000 the correlation reaches high values both between WT and mutant samples (WT1 vs.\ WT2: r=0.73; MUT1 vs.\ MUT2: r=0.98), but we lose a significant proportion of the detected transcripts (465 transcripts retained in the WT samples; 315 transcripts retained in the mutant).

The increase of correlation between mutant and WT biological replicates showed some interesting differences: correlation between the mutant replicates is already high at low coverage thresholds and initially increases with a higher rate; this is, however, coupled with a higher number of discarded transcripts. The higher initial and post-filtering correlation between the mutant replicates likely has both technical and biological causes. First, the tails in mutants are likely to stay closer to their initial length and so are expected to be more homogeneous \cite{traven05,beilharz07}. Second, tails in the mutants are closer to the read length; hence the drift towards longer lengths is expected to have a smaller confounding effect.

The mean correlation between pairs of technical replicates showed a similar behaviour at increasing coverage thresholds in the case of the two mutant samples (\textbf{Figure~4B}), both curves being similar in shape to the change in correlation between mutant biological replicates. This suggests that the mutant samples suffer from comparable amount of technical noise and that this is the likely cause of reduced correlations at lower coverage thresholds.

The mean correlation between technical replicates is substantially lower in the case of WT samples (\textbf{Figure~4B}), which can be explained both by higher intra-transcript variability or higher levels of technical noise. The correlations are especially low in the the case of WT2. This makes it likely that WT2 suffers from the highest levels of technical noise, which can be partly responsible for the lower correlation between WT biological replicates.

Regardless of low correlation at low G-tail fragment coverage, tail run distributions showed excellent consistency between WT biological replicates in the case of highly expressed transcripts (see examples in \textbf{Figure~5}). Most transcripts with high G-tail fragment coverage have longer mean tail runs in the mutant samples (\textbf{Figure~5A,B}), with the magnitude of difference being similar to the difference in global mean tail lengths, However, a few transcripts --- most notably transcript YER177W (BMH1) --- had a short mean tail length in both the WT and mutant samples (\textbf{Figure~5C}). This suggests that poly(A) tail length in this transcript might be under another specific regulatory mechanism, potentially worthy of further investigation.

The consistency of the tail run distributions for transcripts such as those shown in \textbf{Figure~5} suggests that, given sufficient coverage, PAT-seq has the potential to provide a reliable assay of per-transcript mean poly(A) tail length and potentially provide information about the details of the tail length distribution (e.g.~multimodality). However, considering the lower number of high-coverage transcripts and the poor correlation with the PAL-seq assay (see below) we refrain from further downstream analyses of the per-transcript tail run lengths in this experiment.

\subsection*{Correlation between PAT-seq and previous high-throughput assays}

We have examined the correlation of WT mean per-transcript tail run lengths for the 465 transcripts with at least 1000 G-tail fragment coverage with results obtained from \textit{S.~cerevisiae} by the PASTA \cite{beilharz07} and PAL-seq \cite{subtelny14} protocols. There was no correlation between the PAT-seq and PAL-seq results (WT1 vs.\ PAL-seq: $r$=0.07; WT2 vs.\ PAL-seq: $r$=0.03; \textbf{Figure~S2}), despite the high correlation ($r$=0.73) of PAT-seq between biological replicates for the high coverage subset and the high correlation between PAL-seq biological replicates. The correlation between PAL-seq and PASTA results was also weak (r=0.14 for 3631 transcripts; \textbf{Figure~S3}), while correlation between PAT-seq and PASTA was negative (\textbf{Figure~S4}). \note{OK, I'm now happy with this ordering, no need to switch.}

The observed poor correlation between the different assays suggests that the polyadenylation status of the transcriptome might be especially sensitive to growth conditions, or alternatively can be explained by strong and reproducible biases affecting one or more of the assays. 
\note{Text about ranges deleted, as agreed at The Green Man.}
%Also, the PAT-seq high coverage transcript subset mean tail lengths showed a wider range than the corresponding lengths estimated by PAL-seq (\textbf{Figure~S2}).  This suggests that either the PAT-seq method has a greater dynamic range, or that our PAT-seq data were inherently noisier. 
Additional experiments will be necessary to further compare these alternative methods.

\subsection*{The potential of PAT-seq and future directions in studying polyadenylation}

We believe that in the future, as sequencing technologies permit longer and more accurate reads, assays based on direct sequencing of the PAT will be the dominant techniques for studying the biology of polyadenylation. We also believe that PAT-seq has the potential to become the assay of choice on the Illumina platform.  The results of our pilot experiment suggest several ways to improve its properties. First, we have demonstrated that the reliability of the tail length estimates can be increased simply by increasing sequencing depth, suggesting that by using one Illumina lane per sample it might be possible to reliably assess the polyadenylation status of the complete yeast transcriptome. 

Second, adding random barcodes to the sequencing adapter could solve the cluster detection issues and recover many lost G-tail fragments.  It may also be possible to ameliorate sequencing problems caused by the repetitive tail by introducing random mutations in the library as has been suggested in the SAPAS method \cite{fu11} or the NG-SAM approach to sequence assembly \cite{sipos12}.

Third, improvements in Illumina chemistries and base calling algorithms might improve the quality of reads overlapping the tail even without any alterations to our experimental protocol. With one or more of these improvements, the PAT-seq approach could provide a practical alternative to more complicated high throughput assays such as PAL-seq or protocols based on the still relatively expensive third-generation sequencing platforms \cite{sharon13}.

A shortcoming of PAT-seq is that the maximum tail length which can be detected is limited by the read length. This means that for organisms with potentially longer tails (such as mammals), longer reads are needed in order to achieve sufficient dynamic range. The MiSeq Illumina platform is already capable of generating longer reads suitable for this purpose.

One of the most puzzling results of our pilot study is the low correlation between the PAT-seq and PAL-seq assays, even for the set of high-coverage transcripts and despite excellent reproducibility between biological replicates for both methods. While additional experiments are needed to uncover the source of this discrepancy, this observation suggests that the variability of the polyadenylation status of the yeast transcriptome in response to growth and environmental factors is yet to be uncovered.

\section*{Conclusions}

In this study we have presented PAT-seq, a new high-throughput approach to measure poly(A) tail lengths using Illumina short read sequencing. We demonstrated the utility of this approach by estimating and comparing tail lengths from wild type \textit{S.~cerevisiae} and a deadenylase-mutant with expected longer tails. We found that, given high-enough coverage, the tail lengths can be confidently measured even in the presence of ``slippage'' noise.

The distributions of tail run lengths tabulated from the whole wild type and mutant transcriptomes showed excellent consistency between biological replicates, suggesting that PAT-seq is suitable for the assessment of global polyadenylation status in yeast.
Correlation of per-transcript mean tail lengths between biological and technical replicates was high only for transcripts with high G-tail coverage, implying that filtering-out transcripts with low coverage is an essential step in quality control of PAT-seq data.

Based on our results we believe that direct sequencing of poly(A) tails will be the method of choice for studying polyadenylation and that after implementing the suggested improvements PAT-seq has the potential to become an assay of choice for the Illumina platform.
\section*{Acknowledgements}

We would like to thank Prof.\ Thomas Preiss and Stuart Archer for providing the \textit{$\Delta$ccr4-1/pan2} mutant yeast strain, Vicente Pelechano for providing the spike-in standards and helpful discussion in the planning stage and Benjamin Raeder for technical expertise.

BS was funded by a European Molecular Biology Laboratory (EMBL) Interdisciplinary Postdoc (EIPOD) under Marie Curie Actions (COFUND). AMS, GS, TM, JK and NG were supported by EMBL.  GS is a member of Wolfson College, University of Cambridge.

\bibliography{pat_seq} 
\section*{Figures}
\vspace{1cm}

\begin{itemize}
\item[]{
\begin{center}
\includegraphics[scale=1.5]{Figure1.png}
\end{center}

\textbf{Figure 1. Illustration of PAT-seq protocol.}
}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=1.25]{Figure2.png}
\end{center}

\textbf{Figure 2.~The distribution of the number of aligned A bases until the first non-A base (``tail runs'') in the reads mapping to the poly(A) tracts of known length of the spike-in standards.} The vertical green line marks the mean run length from all 53558 reads (44.6), while the red line marks the true length of the spike-in poly(A) tracts (43).}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=0.9]{Figure3.png}
\end{center}

\textbf{Figure 3. Shift in global polyadenylation status in \textit{$\Delta$ccr4-1/pan2} mutants.} \textbf{A}.~Distribution of tail run lengths from all transcripts in the wild type (WT1, blue) and the mutant (MUT1, red) samples. The coloured vertical lines indicate the means of the respective distributions.
\textbf{B}.~The distribution of significant (Mann-Whitney U-test, $p < 0.05$) per-transcript mean tail run length differences (MUT1 - WT1).
}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=0.9]{Figure4.png}
\end{center}

\textbf{Figure 4. Pearson correlation between biological and technical replicates at different G-tail fragment coverage thresholds.} \textbf{A}.~Correlation between biological replicates.  \textbf{B}.~Average correlation between technical replicates from a given sample.}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=0.8]{Figure5.png}
\end{center}

\textbf{Figure 5. Tail run distributions for selected transcripts with high G-tail fragment coverage in WT1 and MUT1 (left) and 
WT2 and MUT2 (right) samples.} \textbf{A.}~\diff{Small heat shock protein HSP26/YBR072W (coverage XXX)}. \textbf{B.}~\diff{HSP12/YFL014W (coverage XXX)}. \textbf{C.}~\diff{BMH1/YER177W (coverage XXX)}.
\diff{Botond to find the coverage values for these three examples:  since they have been selected as good examples where the coverage is high, it is necessary to state what that coverage is.} \note{Well, for the the moment I cannot do this as have no access to the clusters. Also, if I have lost data then this could be potentially cumbersome and require re-running everything :(. So I would ask you to consider letting this slide, one can *kinda* read the coverages from the graphs.}
}

\end{itemize}

\newpage 

\section*{Supporting information}

\begin{itemize}

\item[]{
\note{Not sure what is the best way to include the separate Text\_S1.pdf document in here.  We could put some sort of `see following pages' comment here, and then use a program to insert the pdf pages next?} \\
\textbf{Text S1. Index of selected raw results.}
}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=0.9]{FigureS1.png}
\end{center}

\textbf{Figure S1. Shift in global polyadenylation status in \textit{$\Delta$ccr4-1/pan2} mutants.} \textbf{A.}~Distribution of tail run lengths from all transcripts in the wild type (WT2, in blue) and the (MUT2, in red) mutant samples. The coloured vertical lines mark the means of the respective distributions.
\textbf{B.}~The distribution of significant (Mann-Whitney U-test, $p < 0.05$) per-transcript mean tail run length differences (MUT2 - WT2).
}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=0.9]{FigureS2.png}
\end{center}

\textbf{Figure S2. Correlation of per-transcript mean tail lengths estimated by PAT-seq and PAL-seq in wild type yeast.} \textbf{A.}~Scatter plot and Pearson correlation of PAT-seq mean tail lengths at a coverage threshold of 1000 in WT1 with corresponding results from PAL-seq \cite{subtelny14}. \textbf{B.}~Scatter plot and Pearson correlation of PAT-seq mean tail lengths at a coverage threshold of 1000 in WT2 with PAL-seq.}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=1.0]{FigureS3.png}
\end{center}

\textbf{Figure S3. Scatter plot and correlation between wild type yeast mean tail lengths obtained by PAL-seq \cite{subtelny14} and weighted lengths obtained by the PASTA method \cite{beilharz07}. }
}
\vspace{1cm}

\item[]{
\begin{center}
\includegraphics[scale=0.9]{FigureS4.png}
\end{center}

\textbf{Figure S4. Scatter plot and correlation between wild type yeast mean tail lengths obtained by PAT-seq and weighted lengths obtained by the PASTA method. \textbf{A.}~WT1 vs.\ PASTA. \textbf{B.}~WT2 vs.\ PASTA. }
}

\end{itemize}

\end{document}


